### PR TITLE
Add Record type for FETCH statement in SELECT.

### PR DESCRIPTION
**Pull request by aoisensi**
I don't have good English so please add comments.

### CODE REVIEW

**Comment by aoisensi** [comment]
Maybe `Record.Fetched` is better than `Record.OnlyID`?
Idk.

File: record.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package surrealdb
```
Comment (ElecTwix) [inline record.go:1]: this should be in in other module not in the main surrealdb package

File: record_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+						Bio: "Watch anime Kill Me Baby",
+					},
+				},
+			},
+		},
+	}
+	for _, rule := range rules {
+		t.Run(rule.Name, func(t *testing.T) {
+			var actual testRecordUser
+			err := json.Unmarshal([]byte(rule.Data), &actual)
```
Comment (ElecTwix) [inline record_test.go:65]: shouldn't we also add end to end tests?

File: record_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	"github.com/surrealdb/surrealdb.go"
+)
+
+type testRecordUser struct {
+	ID      string                               `json:"id,omitempty"`
+	Profile surrealdb.Record[*testRecordProfile] `json:"profile,omitempty"`
```
Comment (ElecTwix) [inline record_test.go:13]: why we should have this instead of just basemodel?

**Review by ElecTwix — COMMENTED** [review]
Thanks for PR :)

I'm not sure why we will need something like this.
users already can unmarshal with the marshal package.
I understand the point of the PR I'm not sure it will help.

Could you give us a beneficial example that this PR has provided?

**Comment by aoisensi** [comment]
I had been using the old version and didn't even know it had been moved to `pkg/marshal`.
Apparently `marshal.Basemodel` is sufficient.
I wish they would update the documentation more!
Thank you.

**Comment by aoisensi** [comment]
Oh, v0.2.1 has too.
Bruh...

**Pull request closed without merging**